### PR TITLE
Adding man pages to document ULFM functions 

### PR DESCRIPTION
**Pull request by abouteiller**
Adding man pages that document ULFM functions.

The function are documented in the form that assumes pending PRs #10507 #9925 are merged prior.

### CODE REVIEW

**Comment by jsquyres** [comment]
@abouteiller CI generated Sphinx errors:

```
/home/docs/checkouts/readthedocs.org/user_builds/ompi/checkouts/10781/docs/man-openmpi/man3/MPIX_Comm_agree.3.rst:83: WARNING: undefined label: 'mpi_comm_agree'
/home/docs/checkouts/readthedocs.org/user_builds/ompi/checkouts/10781/docs/man-openmpi/man3/MPIX_Comm_agree.3.rst:79: WARNING: undefined label: 'mpi_comm_agree'
/home/docs/checkouts/readthedocs.org/user_builds/ompi/checkouts/10781/docs/man-openmpi/man3/MPIX_Comm_shrink.3.rst:86: WARNING: undefined label: 'mpi_comm_shrink'
/home/docs/checkouts/readthedocs.org/user_builds/ompi/checkouts/10781/docs/man-openmpi/man3/MPIX_Comm_shrink.3.rst:90: WARNING: undefined label: 'mpi_comm_shrink'
```

**Comment by jsquyres** [comment]
Oh, I just saw your not on the description: will merging https://github.com/open-mpi/ompi/pull/10507 and https://github.com/open-mpi/ompi/pull/9925 fix the Sphinx errors?

**Comment by abouteiller** [comment]
No these were bona fide typos. The reason the PR is dependent upon the other two is that it documents functions that are introduced in them.

**Review by bosilca — APPROVED** [review]

**Comment by abouteiller** [comment]
bot:retest

File: docs/man-openmpi/man3/MPIX_Comm_ack_failed.3.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+WHEN COMMUNICATOR IS AN INTER-COMMUNICATOR
+------------------------------------------
+
+When the communicator is an inter-communicator, the failures of members
+in both the local and the remote groups of *comm* are acknowledged.
+
+ERRORS
+------
+
+Almost all MPI routines return an error value; C routines as the value
```
Comment (jsquyres) [inline docs/man-openmpi/man3/MPIX_Comm_ack_failed.3.rst:124]: Does this text agree with `../ERRORS.rst` (that is included in all other man pages)?

File: docs/man-openmpi/man3/MPIX_Comm_ack_failed.3.rst [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Before the error value is returned, the current MPI error handler is
+called. By default, this error handler aborts the MPI job, except for
+I/O function errors. The error handler may be changed with
+:ref:`MPI_Comm_set_errhandler`; the predefined error handler MPI_ERRORS_RETURN
+may be used to cause error values to be returned. Note that MPI does not
+guarantee that an MPI program can continue past an error.
+
+.. seealso::
+    :ref:`MPIX_Comm_get_failed` :ref:`MPIX_Comm_agree`
```
Comment (jsquyres) [inline docs/man-openmpi/man3/MPIX_Comm_ack_failed.3.rst:135]: We recently changed the style of "See also" to be a bulleted list.  Can you update these pages to match?

E.g., https://github.com/open-mpi/ompi/blame/main/docs/man-openmpi/man3/MPI_Send.3.rst#L85-L88

**Comment by jsquyres** [comment]
Thanks @abouteiller.  Please also cherry pick to v5.0.x.

**Pull request merged**